annotation-server/src/guidelines/guidelines.service.ts

```diff
@@ -0,0 +1,189 @@
+                    'GOOGLESHEET_RANGE_IMPLICATIONS',
+                ),
+                this.configService.get<string>(
+                    'GOOGLESHEET_RANGE_RECOMMENDATIONS',
+                ),
+            ],
+        );
+
+        this.spreadsheetPhenotypeHeader = phenotypeHeader[0].map(
+            (cell) =>
+                new Set(
+                    cell.value
+                        .split(';')
+                        .map((phenotype) => phenotype.trim().toLowerCase()),
+                ),
+        );
+
+        const guidelines: Guideline[] = [];
+        for (let row = 0; row < medications.length; row++) {
+            const geneSymbolName = genes[row]?.[0];
+            const medicationName = medications[row]?.[0];
+            if (!geneSymbolName || !medicationName) continue;
+
+            const medication = await this.findMedication(medicationName.value);
+            const genePhenotypes = await this.findGenePhenotypes(
+                geneSymbolName.value,
+            );
+            if (genePhenotypes.length === 0 || !medication) continue;
+
+            for (let col = 0; col < implications[row].length; col++) {
+                for (const genePhenotype of genePhenotypes[col].values()) {
+                    const guideline = new Guideline();
+                    const implication = implications[row][col].value?.trim();
+                    const recommendation =
+                        recommendations[row][col].value?.trim();
+                    const warningLevel = this.getWarningLevelFromColor(
+                        recommendations[row][col].backgroundColor,
+                    );
+                    if (
+                        !implication ||
+                        implication.replace(' ', '').toLowerCase() === 'n/a' ||
+                        !recommendation ||
+                        recommendation.replace(' ', '').toLowerCase() === 'n/a'
+                    ) {
+                        continue;
+                    }
+                    guideline.implication = implication;
+                    guideline.recommendation = recommendation;
+                    guideline.warningLevel = warningLevel;
+                    guideline.genePhenotype = genePhenotype;
+                    guideline.medication = medication;
+                    guidelines.push(guideline);
+                }
+            }
+        }
+
+        this.guidelinesRepository.save(guidelines);
+
+        this.clearMaps();
+    }
+
+    private getWarningLevelFromColor(
+        color?: sheets_v4.Schema$Color,
+    ): WarningLevel | null {
+        if (!color) return null;
+        const [red, green, blue] = [color.red, color.green, color.blue];
+        if (!red && green === 1 && !blue) return WarningLevel.GREEN;
+        if (red === 1 && green === 1 && !blue) return WarningLevel.YELLOW;
+        if (red === 1 && !green && !blue) return WarningLevel.RED;
+        if (red === green && red === blue && blue === green) return null;
```

Comment:
This means the color can be any shade of gray, e.g. white or black or, most importantly, all colors can be undefined which is the case for transparent / unset backgrounds.

Reply:
that makes sense. Could you maybe add a comment for that? Because I would have to look into this conversation everytime I will be reading this piece of code :sweat_smile: